src/logic/player.h

```diff
@@ -174,36 +174,29 @@ class Player {
 		// Seafaring constants for controlling expeditions
 		static constexpr uint32_t kColonyScanStartArea = 35;
 		static constexpr uint32_t kColonyScanMinArea = 12;
-		static constexpr Time kNoExpedition{0};
+		static constexpr Time kNoExpedition{0U};
 
-		AiPersistentState()
-		   : expedition_start_time(0),
-
-		     magic_numbers(kMagicNumbersSize, 0),
-		     neuron_weights(kNeuronPoolSize, 0),
-		     neuron_functs(kNeuronPoolSize, 0),
-		     f_neurons(kFNeuronPoolSize, 0) {
-		}
+		AiPersistentState() = default;
 
 		void initialize();
 
 		// Was initialized
 		bool initialized{false};
-		uint32_t colony_scan_area{0};
-		uint32_t trees_around_cutters{0};
-		Time expedition_start_time;
+		uint32_t colony_scan_area{0U};
+		uint32_t trees_around_cutters{0U};
+		Time expedition_start_time{0U};
 		int16_t ships_utilization{
 		   0};  // 0-10000 to avoid floats, used for decision for building new ships
 		bool no_more_expeditions{false};
 		int16_t last_attacked_player{0};
 		int32_t least_military_score{0};
 		int32_t target_military_score{0};
-		uint32_t ai_productionsites_ratio{0};
+		uint32_t ai_productionsites_ratio{0U};
 		int32_t ai_personality_mil_upper_limit{0};
-		std::vector<int16_t> magic_numbers;
-		std::vector<int8_t> neuron_weights;
-		std::vector<int8_t> neuron_functs;
-		std::vector<uint32_t> f_neurons;
+		std::vector<int16_t> magic_numbers{kMagicNumbersSize, 0};
```

Comment:
the in initialization needs to fill kMagicNumbersSIze elements of the vector with zeroes. I am not sure whether the braces will do this here. I think this is a one of the few parts where we need to use normal brackets

Reply:
The initializer list in the curly braces is a list of arguments to pass to the type's constructor, so 
```c++
		std::vector<int16_t> magic_numbers{kMagicNumbersSize, 0};
```
is identical to
```c++
		std::vector<int16_t> magic_numbers(kMagicNumbersSize, 0);
```

The syntax to create a vector with the two entries `kMagicNumbersSize` and `0` would be either of
```c++
		std::vector<int16_t> magic_numbers = {kMagicNumbersSize, 0};
		std::vector<int16_t> magic_numbers{{kMagicNumbersSize, 0}};
```
which first creates that vector and then uses it as the default variable value (first variant) or the only argument to the copy constructor (second variant).